### PR TITLE
build(webflow): use node16 module resolution

### PR DESCRIPTION
**Pull request by DLehenbauer**
Semi-automatic PR to switch to node16 module resolution

### CODE REVIEW

File: examples/data-objects/webflow/src/test/layout.spec.ts [hunk 1]
```diff
@@ -58,8 +60,11 @@ describeCompat("Layout", "LoaderCompat", (getTestObjectProvider) => {
 
 	afterEach(() => {
 		layout.remove();
-		layout = undefined;
-		root = undefined;
+
+		// Out of paranoia, force setting to 'undefined' to ensure that 'layout' and 'root'
+		// are re-created between tests.
+		layout = undefined as any;
+		root = undefined as any;
```
Comment (DLehenbauer) [inline examples/data-objects/webflow/src/test/layout.spec.ts:67]: Semantically, the test is replace 'layout' and 'root' before each test.

Forcibly setting these values to undefined after each test is an unnecessary/paranoid way to ensure this happens.

Alternatives to casting to any would be to remove these lines or sprinkle '!' or checks for undefined throughout the tests.

File: examples/data-objects/webflow/tsconfig.json [hunk 2]
```diff
@@ -1,18 +1,11 @@
...
 	"exclude": ["src/test/**/*"],
 	"compilerOptions": {
-		"composite": true,
-		"jsx": "react",
 		"strictNullChecks": false,
 		"rootDir": "./src",
-		// Note: despite being esnext, this publishes to "dist" over "lib" to keep root project settings' debug launch targets
-		// reasonable (source map config takes a dependency on executed JS being in "dist" for mocha tests).
-		// If we convert the remaining group of client packages to use esnext modules only,
-		// we likely want to standardize the esnext build directory.
```
Comment (DLehenbauer) [inline examples/data-objects/webflow/tsconfig.json:12]: We are in the process of standardizing the ESM directory.  :-)

**Review by RishhiB — APPROVED** [review]

**Review by jason-ha — CHANGES_REQUESTED** [review]

File: examples/data-objects/webflow/package.json [hunk 3]
```diff
@@ -132,24 +130,12 @@
 		"browser": {
 			"umd": {
 				"files": [
-					"dist/main.bundle.js"
+					"lib/main.bundle.js"
```
[the remark this answers is not in the document]
Comment (RishhiB) [reply examples/data-objects/webflow/package.json:133]: should we change webpack's config to lib?

File: examples/data-objects/webflow/package.json [hunk 4]
```diff
@@ -41,9 +39,9 @@
 		"start:tinylicious": "webpack serve --config webpack.config.cjs --env mode=tinylicious",
 		"test": "npm run test:mocha",
 		"test:coverage": "c8 npm test",
-		"test:mocha": "cross-env \"NODE_OPTIONS=--experimental-loader esm-loader-css\" mocha --config src/test/.mocharc.cjs \"dist/test/**/*.spec.js\"",
+		"test:mocha": "cross-env \"NODE_OPTIONS=--experimental-loader esm-loader-css\" npm run test:mocha:esm",
+		"test:mocha:esm": "mocha --recursive \"lib/test/**/*.spec.*js\" --exit",
```
[the remark this answers is not in the document]
Comment (DLehenbauer) [reply examples/data-objects/webflow/package.json:43]: Separate is a momentarily a little more convenient for mechanical conversions.

(Except for wierd packages like this.  Good catch re:cross-env.)

File: examples/data-objects/webflow/package.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (jason-ha) [reply examples/data-objects/webflow/package.json]: We should not. webpack should use its own path. I advocate for it to use webpack or some other new distinctive path. @DLehenbauer, did you file a follow-up item for that? I don't see one scanning tasks under [AB#7296](https://dev.azure.com/fluidframework/internal/_workitems/edit/7296).

**Review by jason-ha — APPROVED** [review]

**Pull request merged**